File: 0014-longest-common-prefix/solution.go
func longestCommonPrefix(strs []string) string {
    n := len(strs)
    if n == 1 {
        return strs[0]
    }

    prefix := []rune(strs[0])
    for i := 1; i < n; i++ {
        str := strs[i]

        for j := 0; j < len(prefix); j++ {
            if j >= len(str) || rune(str[j]) != prefix[j] {
                prefix = prefix[:j]
                break
            }
        }
    }

    return string(prefix)
}

File: 0014-longest-common-prefix/solution.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let result = "";
    for (let i = 0; i < strs[0].length; i++) {
        let allSame = true;
        for (let j = 1; j < strs.length; j++) {
            if (i >= strs[j].length || strs[j][i] !== strs[0][i]) {
                allSame = false;
                break;
            }
        }

        if (allSame) {
            result += strs[0][i];
        } else {
            break;
        }
    }

    return result;
};
